### DISCUSSION TITLE
Can you support unselecting listbox items.

### DISCUSSION THREAD

**Discussion by DarraghOFlaherty** [tailwindlabs/headlessui#1817] [Ideas]
Once you select a list box item you cannot unselect the list box item you can only change your selection to another item. It would be nice to have the option to click on the currently selected option to unselect it.

**Comment by JesusTheHun** [comment]
The idea of the listbox is to replicate the behavior of a native select. Which cannot be unselected. 
You are free to add an option to represent an empty value though.

**Comment by jdalrymple** [comment]
Seems odd then that the multi select allows for you to select 0->1+ (you can unselect all options) but the single select requires 1 (you cant unselect an option)

**Comment by jdalrymple** [comment]
Another way to support this is adding a max selected prop for the multiselect, and setting that to 1

**Comment by drochag** [comment]
So @JesusTheHun if the idea of the listbox is to replicate the behavior of a native select, having this additional empty option
```
const customOptions: Option[] = useMemo(
    () => [
      {
        id: 0,
        name: 'Select an option',
        value: '',
      },
      ...options,
    ],
    [placeholder, options],
  )
```
and having the default value as `''` it should select the `Select an option` option, instead it shows nothing, where a native select would select the empty option, right?

![Screenshot 2024-04-01 at 10 05 48](https://github.com/tailwindlabs/headlessui/assets/1919322/6ee257a5-f487-4430-aa70-ab50860b6fbe)
![Screenshot 2024-04-01 at 10 05 52](https://github.com/tailwindlabs/headlessui/assets/1919322/8379a26b-d1f1-410a-b75f-aa34d35e941c)

Note that we are correctly having the null / empty value mapped on the HTML for the hidden input

![Screenshot 2024-04-01 at 10 07 44](https://github.com/tailwindlabs/headlessui/assets/1919322/3d9c7085-c07c-4c53-98ca-9670b74a7963)

And when selecting the created empty option the input with the value doesn't change either (although the other properties of the object are added)

![Screenshot 2024-04-01 at 10 09 12](https://github.com/tailwindlabs/headlessui/assets/1919322/faeb2994-e2b9-4219-abb6-9b7035d166dd)
